jwst/assign_wcs/nirspec.py

```diff
@@ -1696,6 +1696,129 @@ def gwa_to_ymsa(msa2gwa_model, lam_cen=None, slit=None, slit_y_range=None):
     return tab
 
 
+def get_transforms(input_model, slitnames, return_slits=False):
+
+    """
+    Returns a WCS object together with the transforms needed for all
+    slits.  This enables the code to avoid excessive deep copying of
+    of WCS objects.
+
+    Parameters
+    ----------
+    input_model : `~jwst.datamodels.JwstDataModel`
+        A data model with a WCS object for the all open slitlets in
+        an observation.
+    slitnames : int or str
+        Slit.name of all open slits.
+    return_slits : bool
+        Return the open slits
+
+    Returns
+    -------
+    wcsobj : `~gwcs.wcs.WCS`
+        WCS object deep copied from input_model.meta.wcs
+    trans1 : `~astropy.modeling.core.Model` model
+        wcs.pipeline[1].transform[1:]
+    trans2 : list of `~astropy.modeling.core.Model` models
+        wcs.pipeline[2].transform model for each input slit
+    trans3 : list of `~astropy.modeling.core.Model` models
+        wcs.pipeline[3].transform model for each input slit
+    open_slits : list of slits
+        open slits from wcs.get_transform('gwa', 'slit_frame').slits
+        Only returned if return_slits is True
+    """
+    
+    import copy
```

Comment:
Is there a reason for this to be a lazy import, or can it go at the top of the module?  It's imported in several places in this module.

Reply:
It was the approach used in _nrs_wcs_set_input before I started editing; I copied that not knowing if it was preferred in this case.  I am happy to move it to the top.  It doesn't look like it will cause any problems with copy used a variable name, for example.

Reply:
I thought that might be the case.  Let's go ahead and fix it, then.

Reply:
Fixed.